File: stream-chat-android-docs/src/main/java/io/getstream/chat/docs/java/client/docusaurus/HandlingUserConnection.java
package io.getstream.chat.docs.java.client.docusaurus;

import io.getstream.chat.android.client.ChatClient;
import io.getstream.chat.android.models.User;

/**
 * @see <a href="https://getstream.io/chat/docs/sdk/android/client/guides/channel-list-updates/">Channel List Updates</a>
 */
public class HandlingUserConnection {

    public void connectingAUser() {
        User user = new User.Builder()
                .withId("bender")
                .withName("Bender")
                .withImage("https://bit.ly/321RmWb")
                .build();

        // Check if the user is not already set
        if (ChatClient.instance().getCurrentUser() == null) {
            ChatClient.instance().connectUser(user, "userToken")  // Replace with a real token
                    .enqueue((result) -> {
                        if (result.isSuccess()) {
                            // Handle success
                        } else {
                            // Handle error
                        }
                    });
        }
    }

    public void disconnectTheUser() {
        boolean flushPersistence = false;
        ChatClient.instance().disconnect(flushPersistence).enqueue((result) -> {
            if (result.isSuccess()) {
                // Handle success
            } else {
                // Handle error
            }
        });
    }

    public void switchTheUser() {
        User user1 = new User.Builder()
                .withId("bender")
                .withName("Bender")
                .withImage("https://bit.ly/321RmWb")
                .build();

        // Connect the first user
        ChatClient.instance().connectUser(user1, "userToken") // Replace with a real token
                .enqueue((result) -> {
                    if (result.isSuccess()) {
                        // Handle success
                    } else {
                        // Handle error
                    }
                });

        User user2 = new User.Builder()
                .withId("bender2")
                .withName("Bender2")
                .withImage("https://bit.ly/321RmWb")
                .build();
        ChatClient.instance().switchUser(user2, "userToken") // Replace with a real token
                .enqueue((result) -> {
                    if (result.isSuccess()) {
                        // Handle success
                    } else {
                        // Handle error
                    }
                });
    }
}

File: stream-chat-android-docs/src/main/kotlin/io/getstream/chat/docs/kotlin/client/docusaurus/HandlingUserConnection.kt
package io.getstream.chat.docs.kotlin.client.docusaurus

import io.getstream.chat.android.client.ChatClient
import io.getstream.chat.android.models.User
import io.getstream.chat.android.client.utils.Result

/**
 * @see <a href="https://getstream.io/chat/docs/sdk/android/client/guides/channel-list-updates/">Channel List Updates</a>
 */
class HandlingUserConnection {

    fun connectingAUser() {
        val user = User(
            id = "bender",
            name = "Bender",
            image = "https://bit.ly/321RmWb",
        )

        // Check if the user is not already set
        if (ChatClient.instance().getCurrentUser() == null) {
            ChatClient.instance().connectUser(user = user, token = "userToken") // Replace with a real token
                .enqueue { result ->
                    when (result) {
                        is Result.Success -> {
                            // Handle success
                        }
                        is Result.Failure -> {
                            // Handler error
                        }
                    }
                }
        }
    }

    fun disconnectTheUser() {
        ChatClient.instance().disconnect(flushPersistence = false).enqueue { result ->
            when (result) {
                is Result.Success -> {
                    // Handle success
                }
                is Result.Failure -> {
                    // Handler error
                }
            }
        }
    }

    fun switchTheUser() {
        val user1 = User(
            id = "bender",
            name = "Bender",
            image = "https://bit.ly/321RmWb",
        )

        // Connect the first user
        ChatClient.instance().connectUser(user = user1, token = "userToken") // Replace with a real token
            .enqueue { result ->
                when (result) {
                    is Result.Success -> {
                        // Handle success
                    }
                    is Result.Failure -> {
                        // Handler error
                    }
                }
            }

        val user2 = User(
            id = "bender2",
            name = "Bender2",
            image = "https://bit.ly/321RmWb",
        )

        ChatClient.instance().switchUser(user = user2, token = "userToken") // Replace with a real token
            .enqueue { result ->
                when (result) {
                    is Result.Success -> {
                        // Handle success
                    }
                    is Result.Failure -> {
                        // Handler error
                    }
                }
            }
    }
}
